code/espurna/sensors/ADE7953Sensor.h

```diff
@@ -0,0 +1,215 @@
+                active_power1 = (int32_t)Ade7953Read(0x313) * -1;  // Relay 1
+                active_power1 = (active_power1 > 0) ? active_power1 : 0;
+            }
+            current_rms2 = Ade7953Read(0x31A);     // Relay 2
+            if (current_rms2 < 2000) {             // No load threshold (20mA)
+                current_rms2 = 0;
+                active_power2 = 0;
+            } else {
+                active_power2 = (int32_t)Ade7953Read(0x312);  // Relay 2
+                active_power2 = (active_power2 > 0) ? active_power2 : 0;
+            }
+            _voltage = (float) voltage_rms / ADE7953_UREF;            
+            
+            writeFloat(
+                ADE7953_ALL_RELAYS, 
+                (float)(current_rms1 + current_rms2) / (ADE7953_IREF * 10), 
+                (float)(active_power1 + active_power2) / (ADE7953_PREF / 10)
+            );
+            writeFloat(
+                ADE7953_RELAY_1, 
+                (float) current_rms1 / (ADE7953_IREF * 10), 
+                (float) active_power1 / (ADE7953_PREF / 10)
+            );    
+            writeFloat(
+                ADE7953_RELAY_2, 
+                (float)current_rms2 / (ADE7953_IREF * 10), 
+                (float)active_power2 / (ADE7953_PREF / 10)
+            );
+        }
+
+        void writeFloat(unsigned int relay, float current, float power) {
+            auto& reading_ref = _readings.at(relay); 
+            reading_ref.current = current;
+            reading_ref.power = power;
+        }
+
+        // Current value for slot # index
+        double value(unsigned char index) {        
+            if (index == 0) return _voltage;            
+            int relay = (index - 1) / ADE7953_TOTAL_DEVICES;	
+            index = index % ADE7953_TOTAL_DEVICES;
+            if (index == 0) return _readings[relay].energy;
+            if (index == 1) return _readings[relay].current;
+            if (index == 2) return _readings[relay].power;     
+            return 0;
+        }
+
+    protected:  
+        void _init() {                     
+            delay(100);                     // Need 100mS to init ADE7953
+            Ade7953Write(0x102, 0x0004);    // Locking the communication interface (Clear bit COMM_LOCK), Enable HPF
+            Ade7953Write(0x0FE, 0x00AD);    // Unlock register 0x120
+            Ade7953Write(0x120, 0x0030);    // Configure optimum setting       
+                    
+            _ready = true;
+        }
+
+        int Ade7953RegSize(uint16_t reg) {
+            int size = 0;
+            switch ((reg >> 8) & 0x0F) {
+                case 0x03:
+                size++;
+                case 0x02:
+                size++;
+                case 0x01:
+                size++;
+                case 0x00:
+                case 0x07:
+                case 0x08:
+                size++;
+            }
+            return size;
+        }
+            
+        void Ade7953Write(uint16_t reg, uint32_t val) {
```

Comment:
just `write`?
same thing about the static as above

Reply:
I have problems with static methods because I'm using `_address` from `I2CSensor`. I'm going to add an extra parameter to send the address too